doc/arch/adr-002-rest-api.md

```diff
@@ -0,0 +1,74 @@
+# Boundary Sync REST API Design
+
+## Context
+
+Based on the [data model](./adr-001-data-models.md), the following REST API design is proposed to access it:
+
+### Authentication
+
+| Path            | Method | Body      | Authentication | Responses                                             | Notes                                        |
+| --------------- | ------ | --------- | -------------- | ----------------------------------------------------- | -------------------------------------------- |
+| `/auth/login/`  | GET    | -         | Allow Any      | 200 OK,<br />401 Not Authorized                       | Returns loging JSON if current session valid |
+| `/auth/login/`  | POST   | `{ ... }` | Allow Any      | 200 OK,<br />400 Bad Request,<br />401 Not Authorized | Used for starting new valid session          |
+| `/auth/logout/` | POST   | -         | Logged In User | 200 OK                                                | Used for logging out                         |
+
+### Boundaries
+
+| Path                                       | Method | Body                     | Authentication | Responses                                                  | Notes                                                                                                              |
+| ------------------------------------------ | ------ | ------------------------ | -------------- | ---------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ |
+| `/boundaries/`                             | GET    | -                        | Logged In User | 200 OK                                                     | Returns list of all boundaries the user has access to                                                              |
+| `/boundaries/?utilities=1,3,5,8`           | GET    | -                        | Logged In User | 200 OK,<br />401 Not Authorized                            | Returns list of boundaries in the given utilities. If the user does not have access to that utility, returns a 401 |
+| `/boundaries/`                             | POST   | `{ ... }`                | Contributor    | 201 Created,<br />400 Bad Request,<br />401 Not Authorized | Creates a new boundary if the payload is correct and the user is authorized                                        |
+| `/boundaries/{id}/`                        | GET    | -                        | Logged In User | 200 OK,<br />404 Not Found                                 | Returns boundary details if the users has access to it, else 404s. Includes all details about the boundary.        |
+| `/boundaries/{id}/shape/`                  | PUT    | `{ ... }`                | Contributor    | 200 OK,<br />404 Not Found                                 | Updates a boundary's shape                                                                                         |
```

Comment:
Wondering if this could just be a PATCH method on boundaries/id ?

Reply:
It could, but PATCHes (used for partial updates) are harder to reason about than PUTs (full updates). If a Boundary had multiple editable fields, I would be more open to that, but since the shape is the only thing that can be edited in a boundary, and it may require some special handling (validating the shape, etc), I'm on the side of a dedicated endpoint.